### ISSUE TITLE
New user unfriendly documentation

### ISSUE THREAD

**Issue by evanfields** [tbreloff/Plots.jl#445]
I'm keenly aware how fraught it is to say "I just tried your super complicated software for the first time, here are problems," so I will tread carefully! And I really am impressed by what I've seen of Plots via the documentation. That said, I think the following are worth reporting:

- (This may be a PlotlyJS issue, not Plots) Installation instructions for Plots with PlotlyJS seem insufficient. After `Pkg.add("Plots")` and `Pkg.add("PlotlyJS")`, 
```
julia> using Plots

julia> plot(rand(10), rand(10))
[Plots.jl] Initializing backend: plotlyjs
Error showing value of type Plots.Plot{Plots.PlotlyJSBackend}:
ERROR: Cannot find Electron. Try `AtomShell.install()`.
 in electron at C:\Users\Evan\.julia\v0.4\Blink\src\AtomShell\process.jl:49
```
Of course `AtomShell.install()` gives `ERROR: UndefVarError: AtomShell not defined`. From searching online for something like "electron not found plots.jl" I found an issue with the code `Blink.AtomShell.install()` and that worked, which is a well-duh given the full error message. But searching through old Github issues or parsing error messages should not be part of the install process!

- The documentation does not contain a clear list of the different plot types available. For example, say I'm a new user and I want to make a scatter plot. I might first look in the 'basics' section of the documentation and learn that I should try `plot(...)`. I'll get a line plot, not scatter. Elsewhere in the documentation there are examples of advanced behavior using `scatter(...)`, so that's discoverable, but it would be really nice to see a table with columns "If you want to make a...", "Use function...", "With required arguments...". There are a lot of great examples under PyPlot, PlotlyJS, GR, and PGFPlots, and those help, but are perhaps insufficient on their own. 

Whether it's better for the documentation to have exhaustive examples or a table describing available functions as suggested above is perhaps a personal question. At the least, suppose I wanted to do something with hex bins. That's not in the examples, so the only way I would discover it is by looking in the supported attributes page of the documentation?

- The documentation isn't clear about what kinds of save operations will actually work. I like pdf output for vector graphics and easy inclusion in TeX documents. After getting Electron working and making a simple scatter plot:
```
julia> savefig("test.pdf")
WARNING: _writemime is not defined for this backend. m=application/pdf
```
Okay, so not pdf. I think I remember from last time I gave Plots a try that it doesn't work with all backends. Bummer, but let's see what else we've got. Maybe something that's not a vector graphics format?
```
julia> savefig("test.png")
ERROR: You need to install the Rsvg package use this routine for file type png
Try insalling with `Pkg.add("Rsvg")`
 in error at error.jl:21
```
Okay, maybe svg?
```
julia> savefig("test.svg")
```
Yes! No error. Except the svg file written is unopenable by Inkscape, and Opera just shows some XML rather than an image.

Also interesting is that the png and pdf calls created empty files, which could be tricky for folks running scripts which create plots and check for file existence.

My point here isn't that every backend must work with every output format - I've no doubt that's fiendishly complicated and perhaps not even possible in principle given differences in how backends render, image formats store information, etc. But the documentation should reflect what works, and things should not fail silently.

Anyway, it's clear Plots is incredibly powerful and flexible. But `n=1` it's confusing to a new user, even one who's been using Julia for a while.

**Comment by tbreloff** [comment]
Awesome. Since you've put so mich time and thought into this, I'm excited
that you'll help out by adding and improving the documentation! Head over
to PlotDocs.jl and add what you think is missing/needed with a PR. There's
already a meta-issue for documentation with many of these points... I'll
find it and link to it when I'm in front of a computer.

**Comment by tbreloff** [comment]
To summarize your post:

- There should be install instructions for each backend. I agree, and it
might be best to add this to the "backends" page in each section. In
addition to the electron install for PlotlyJS, many people run into issues
with PyPlot/Conda. It would be good to summarize some troubleshooting notes

- need descriptions of series types. Agreed it's one of the few un-checked
todo items on the documentaion meta-issue

- summaries of supported output formats. Could be another table in the
"supported" page? We'd have to actually compile this information of course

Any help with any of this is very appreciated! There's only so many hours
in the day...

**Comment by tbreloff** [comment]
The documentation meta-issue is #83.  Looking forward to your much needed help @evanfields!

**Comment by evanfields** [comment]
Thanks @tbreloff. Right now I'm ill equipped to add to the docs since I know nothing about Plots, but as I learn more I'll try to update gaps I see. Just update the corresponding `.md` files in PlotDocs and submit a PR?

**Comment by tbreloff** [comment]
Correct Evan. Just update the markdown in a PR. Then I'll build and the
docs myself. I think you're better equipped than me... Sometimes its hard
to write docs for beginners when you're not a beginner.

**Comment by rand5** [comment]
I agree with many of the comments from @evanfields above. Personally, I am a fan of the gallery style doc along the lines of what [PGFplots](http://pgfplots.sourceforge.net/gallery.html) has implemented. It makes it easy to quickly grasp the capabilities of the package while also providing explicit code to recreate examples. I'd be happy to work on the docs and make them more amenable to beginners such as myself.

**Comment by ChrisRackauckas** [comment]
Plots.jl has a gallery-style documentation for each backend. Click on "PyPlot" under examples (it's the page I tend to use most). Is it not clear that's where the examples would be?

**Comment by rand5** [comment]
No, I think it is certainly clear that that is where the examples are, I just don't feel that the examples effectively illustrate all of the capabilities and common use-cases.

**Comment by ChrisRackauckas** [comment]
What's are some things that are left out of the examples which should be included?

**Comment by tbreloff** [comment]
> effectively illustrate all of the capabilities 

I simply don't have the time for that!! :)

**Comment by ChrisRackauckas** [comment]
> effectively illustrate all of the capabilities

Plots.jl has infinite capabilities.

**Comment by rand5** [comment]
@tbreloff Understood- and I certainly didn't mean that as a negative critique. I'll take some time to put together a few basic examples that I feel might be helpful for someone who is new to working with Plots.jl

**Comment by mkborregaard** [comment]
I've updated the title to the actionable elements. @evanfields will you correct me if it doesn't reflect the intention?

**Comment by evanfields** [comment]
Yeah seems about right. Thanks!

**Comment by BeastyBlacksmith** [comment]
The only point that is still open is installation instructions for the backends isn't it?
